File: songchengwen/UtilizationCollection/server/src/main/java/com/h_s/entity/Metric.java
package com.h_s.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "metric")
public class Metric {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "metric")
    private String metric;
    @Column(name = "endpoint")
    private String endpoint;
    @Column(name = "timestamp")
    private long timestamp;
    @Column(name = "step")
    private int step;
    @Column(name = "value")
    private double value;
}

File: huangqinghua/xiaomi_go/server/internal/model/metric.go
package model

type Metric struct {
	Metric    string
	Endpoint  string
	Timestamp int64
	Step      int64
	Value     float64
}

type MetricData struct {
	Metric    string
	Timestamp int64
	Value     string
}

// MetricResponse 查询的回报实体
type MetricResponse struct {
	Metric string
	Data   []ValueData
}

type ValueData struct {
	Timestamp int64
	Value     string
}
